Fix/crud municipes

antoniojefferson opened the pull request:
Aplicando PR de melhorias no CRUD de municipe, com intuito de melhorar o código seguindo os conceitos e padrões rails

antoniojefferson committed:
refactor(crud_municipes): refatorando o codigo, seguindo o padrao e descartando a forma grotesca

Foi realizado um refatoramento do codigo, para seguir os conceitos de cadastro de dois models
relacionados em um mesmo controller e no mesmo processo, foi corrigido o codigo grotesco e malfeito
que era usado antes.

antoniojefferson committed:
refactor(model_adress): corrigindo o nome do relacionamento belongs_to

Corrigindo a referencia belongs_to que antes estava apontando para o nome da tabela no singular

antoniojefferson committed:
refactor(model_municipe): corrigindo o nome do relacionamento belongs_to

Corrigindo a referencia belongs_to que antes estava apontando para o nome da tabela no singular

antoniojefferson committed:
feat(gemfile): adicionando gem para debug

Adicionando a gem pry-rails, para facilitar o debug do codigo em execucao

antoniojefferson committed:
feat(package.json): adicionando commitizen ao projeto

antoniojefferson committed:
refactor(.gitignore): adicionando a referencia da pasta node_modules

Setando a pasta do node_modules, para que a mesma seja ignorada, uma vez que ela so é nescessaria em
desenvolvimento, por nescessidade dos recursos do commitizen

marcelohenriq99 commented on app/models/adress.rb:
```diff
@@ -1,5 +1,5 @@
 class Adress < ApplicationRecord
-  belongs_to :municipe
+  belongs_to :municipes, optional: true
```
acho que era pra ser `belongs_to :municipe` não?

antoniojefferson replied:
Sim, vacilo, vou estar corrigindo na PR.

ricardopacheco replied:
1. Nome de classe incorreto. `Address`
2. Há uma dissonância linguística aqui: Porque o endereço está em en_US e o municipe em pt_BR? Use um ou outro por razões de padronização (a maioria dos projetos usa en por padrão, pra ficar com sonância com a linguagem de programação, outros preferem usar lingua nativa para facilidade de entendimento e comunicação (saudesimples usa essa escolha). Ambas estão corretas, mas, misturar linguagens gera esse efeito dissonântico).

antoniojefferson replied:
Ok, falha minha, eu me enrolei nessas nomeações, a intenção era para ser tudo em ingles, logo depois dessa PR, vu criar uma para corrigir esses nomes e deixar tudo no padrão!

antoniojefferson replied:
Pode ser assim?

ricardopacheco replied:
Claro, o padrão fica a sua escolha. Porém, depois que escolhido, deve ser seguido! ;)

marcelohenriq99 commented on app/controllers/municipes_controller.rb:
```diff
@@ -90,6 +46,9 @@ def find_municipe
+		params.permit(
+      :full_name, :cpf, :cns, :email, :birth_date, :phone, :photo, :status,
+      adresses_attributes: [ :id, :cep, :logradouro, :complement, :district, :city, :uf, :ibge_code ]
+    )
```
Boa!

marcelohenriq99 commented on app/controllers/municipes_controller.rb:
```diff
@@ -53,28 +30,7 @@ def create
+		if @municipe.update(municipe_params)
 			render json: @municipe, status: :ok
 		else
 			render json: { errors: @municipe.errors.full_messages }, status: :unprocessable_entity
```
Verifica se tem I18n para essas mensagens de erro, caso tenha que exibir essas mensagens em tela fica mais fácil.

antoniojefferson replied:
Vou estar verificando!

marcelohenriq99 commented on app/models/municipe.rb:
```diff
@@ -1,21 +1,22 @@
 class Municipe < ApplicationRecord
-    has_many :adress, foreign_key: "municipe_id"
+    has_many :adresses, foreign_key: "municipe_id"
```
Precisa dessa `foreign_key` ?

italopires replied:
Por que o nome do Model está em portugues? Não tinha que ser Citizen?

antoniojefferson replied:
Imagino que não, eu havia seguindo um padrão enquanto estudava um exemplo, deixei por ficar expicito o nome do campo da foreygn_key, ms de fato não faz muito sentido pelo fato de o nome estar igual ao que ele vai criar automaticamente!

antoniojefferson replied:
Vou remover também!

marcelohenriq99 commented on app/models/municipe.rb:
```diff
@@ -1,21 +1,22 @@
+        phone: self.phone,
+        photo: self.photo,
+        status: self.status,
+        adress: self.adresses,
```
não seria mais correto `addresses: self.addresses` ?

antoniojefferson replied:
Por enquanto ficou assim, deixei essa correção para o proximo fluxo onde vou tratar os retornos de dados, nesse ponto vou estar usando serializer para serializar os dados e remover essa função de dentro do model.

marcelohenriq99 commented on app/models/municipe.rb:
```diff
@@ -1,21 +1,22 @@
+        cpf: self.cpf,
+        cns: self.cns,
+        email: self.email,
+        birth_date: self.birth_date.strftime('%d/%m/%Y'),
```
da pra usar um decorator aqui ? o que acontece se não tiver esse dado(birth_date)?

antoniojefferson replied:
Vou cuidar desse ponto no fluxo a seguir!

italopires commented on app/controllers/municipes_controller.rb:
```diff
@@ -18,32 +18,9 @@ def show
-			photo: municipe_params[:photo],
-			status: municipe_params[:status]
-		})
+    
```
ajustaria a identação para 2 espaços

antoniojefferson replied:
Esse ponto já foi corrigido, já não existe mais no códio esse bloco de código onde você viu mais de dois espaços na identação!

italopires commented on app/models/adress.rb:
```diff
@@ -12,7 +12,7 @@ def json
 			created_at: self.created_at,
 			updated_at: self.updated_at,
-			municipe: self.municipe,
+			municipe: self.municipes,
```
Não fazemos json assim. Utilizando Serializers, dá uma pesquisada por ActiveModel::Serializer

ricardopacheco replied:
Porque está pluralizado se é uma relação 1xN?

antoniojefferson replied:
Sim, porem estou seguindo um fluxo de auterações em camadas, primeiro corrigindo as migrations, depois os inserts e updates, logo depois o tratamento de retorno de dados, onde vou estar destrindo essas funções e passando a responsabilidade para o serializer

antoniojefferson replied:
Ricardo esse ponto que voce citou, já foi corrigido, foi uma falha de entendimento meu enqunto discutia umas ideias com o Marcelo aqui!

ricardopacheco commented on Gemfile:
```diff
@@ -32,6 +32,7 @@ group :development, :test do
   gem 'sqlite3', '~> 1.4'
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem 'pry-rails'
```
Ideal é sempre setar a versão específica ou range. É considerado boa prática (quando é necessário atualizar versões major do projeto por exemplo).

antoniojefferson replied:
Certo, vou estar adicionando a versão junto a gem, vale pelo toke nesse ponto!

ricardopacheco commented on app/controllers/municipes_controller.rb:
```diff
@@ -18,32 +18,9 @@ def show
-			status: municipe_params[:status]
-		})
+    
+		@municipe = Municipe.new(municipe_params)
```
👍

antoniojefferson committed:
refactor(adress.rb): deixando a referencia do belongs_to em minusco

Anteriormente foi mudado o name do model no belongs_to no adress para plural, porem foi uma mudança
errada, não interfriu n excução do sistema, porem não estava correto.

marcelohenriq99 commented on package-lock.json:
```diff
@@ -0,0 +1,2624 @@
+{
```
Cuidado com esses arquivos `package.json` e `lock.json` se não for importante pro projeto não sobe.

antoniojefferson replied:
Eles foram criados na adição do commitizen, por isso os subi, nesse caso não precisaria subir?

marcelohenriq99 replied:
não precisa subir esses arquivos eles são apenas para padronizar os teus commits.

antoniojefferson committed:
refactor(municipe.rb): refatorando o uso do nome da foreign_key

Removendo a declaração de name para foreign_key

marcelohenriq99 commented on package.json:
```diff
@@ -0,0 +1,10 @@
+{
```
mesma coisa aqui.

antoniojefferson committed:
refactor(gemfile): adicionando a versão da gem pry-rails, deixando a versão fixa no arquivo!

antoniojefferson committed:
refactor(.gitignore): garantindo que o package.json e package-lock.json, sejam ignorados!

antoniojefferson committed:
refactor(package.json): Removendo o package.json e package-lock.json para garantir que a PR atual não os mantenham

PedroPedrassolli commented on Gemfile:
```diff
@@ -32,6 +32,7 @@ group :development, :test do
   gem 'sqlite3', '~> 1.4'
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem 'pry-rails', '~> 0.3.9'
```
Uma dica apenas, agora que está usando o pry, acredito que possa se livrar do byebug

marcelohenriq99 replied:
olha ai

antoniojefferson replied:
Show, Pedro, o Marcelo fez o memso comentario, acabei de subir um push para remover byebug

PedroPedrassolli replied:
🚀 show!

PedroPedrassolli replied:
@antoniojefferson só não se esquece de commitar depois de rodar o `bundle install`, para que as mudanças reflitam também no `gemfile.lock`

antoniojefferson replied:
Verdade, vou atualizar aqui!

antoniojefferson committed:
refactor(gemfile): removendo a gem byebug que não está sendo usada!

marcelohenriq99 reviewed (approved):
Se atentando aos pontos para o próx PR pra mim ta shows :guitar:

antoniojefferson committed:
refactor(adress.rb): corrigindo a referencia de municipe no model adress
